### PR TITLE
Text normalization

### PR DESCRIPTION
**Pull request by yzhang123**

### CODE REVIEW

File: tools/text_normalization/normalize.py [hunk 1]
```diff
@@ -41,6 +61,24 @@
     tag_verbatim,
 ]
 
+verbalizers = {
+    TagType.CARDINAL: [expand_cardinal, expand_roman],
```
Comment (ekmb) [inline tools/text_normalization/normalize.py:65]: Could you please change `taggers` to TAGGERS and verbalizers -> VERBILIZERS?
Maybe define them in the tagger.py and just import here?

File: tools/text_normalization/tagger.py [hunk 2]
```diff
@@ -63,21 +52,23 @@ def default(self, obj):
 
 class Tag:
     """
-    Class for tagger object that is created for each detected unnormalized token.
+    Class for tagger object that is created for each detected semiotic token.
```
Comment (ekmb) [inline tools/text_normalization/tagger.py:55]: Maybe move Tag and TagType classes to a separate file for better readability?

File: tools/text_normalization/verbalizer.py [hunk 3]
```diff
@@ -55,14 +84,20 @@ def expand_whitelist(data: dict) -> str:
     return _whitelist_dict[data["value"]]
 
 
+def expand_decimal(data: dict) -> str:
+    return _inflect.number_to_words(data["value"]).replace("-", " ").replace(" and ", " ").replace(",", "")
```
Comment (ekmb) [inline tools/text_normalization/verbalizer.py:88]: docstrings are missing

File: tools/text_normalization/verbalizer.py [hunk 4]
```diff
@@ -155,7 +214,7 @@ def _expand_hundreds(text):
         return _inflect.number_to_words(text)
 
 
-def _expand_currency(data):
+def _expand_currency(data: dict) -> str:
```
Comment (ekmb) [inline tools/text_normalization/verbalizer.py:217]: docstring

**Review by ekmb — CHANGES_REQUESTED** [review]
Could you please fix LGTM issues and add a third party header we discussed offline?

File: tools/text_normalization/normalize.py [hunk 5]
```diff
@@ -77,7 +128,7 @@ def select_tags(tags: List[Tag]) -> List[Tag]:
 
 def verbalizer(text: str, tags: List[Tag]) -> str:
     """
-    Given text and corresponding list of tags. Applies verbalizations for tagged substrings and return transduced text.
+    Given text and corresponding list of tags. Applies verbalization where possible for tagged substrings and returns transduced text.
```
Comment (ekmb) [inline tools/text_normalization/normalize.py:131]: Maybe rename this method to verbalize() and the current verbalize() changed to _verbalize() or verbalize_tag?

File: tools/text_normalization/normalize.py (hunk 1 above)
Comment (yzhang123) [reply tools/text_normalization/normalize.py:65]: there would be an import loop, but I will Capitalize them

File: tools/text_normalization/normalize.py [hunk 6]
```diff
@@ -17,6 +17,7 @@
 
 from tools.text_normalization.tagger import (
     Tag,
+    TagType,
```
Comment (ekmb) [inline tools/text_normalization/normalize.py:20]: Should this be `from tools.text_normalization.tag ...`?

File: tools/ctc_segmentation/scripts/prepare_data.py [hunk 7]
```diff
@@ -21,9 +21,9 @@
 from typing import List
 
 import scipy.io.wavfile as wav
-from normalization_helpers import LATIN_TO_RU, RU_ABBREVIATIONS
 from num2words import num2words
 
+from .normalization_helpers import LATIN_TO_RU, RU_ABBREVIATIONS
```
Comment (ekmb) [inline tools/ctc_segmentation/scripts/prepare_data.py:26]: please revert

**Review by ekmb — APPROVED** [review]

**Pull request merged**